File: csharpcore/GildedRose/ItemType.cs
namespace GildedRoseKata;

public static class ItemType
{
    public enum ItemKey
    {
        Regular,
        Legendary,
        BetterWithAge,
        BackstagePasses,
        Conjured
    }
    
    public static bool IsLegendaryItem(Item item) => IsLegendaryItem(item.Name);
    public static bool IsLegendaryItem(string name) => name.ToLower().Contains("sulfuras");

    public static bool IsBackstagePassesItem(Item item) => item.Name.ToLower().Contains("backstage passes");

    public static bool IsBetterWithAgeItem(Item item) => item.Name.ToLower().Equals("aged brie");

    public static bool IsConjuredItem(Item item) => item.Name.ToLower().Contains("conjured");
}

File: Java/src/main/java/com/gildedrose/domain/item/ItemType.java
package com.gildedrose.domain.item;

import java.util.Arrays;

public enum ItemType {

    STANDARD("Standard item"),
    AGEG_BRIE("Aged Brie"),
    BACKSTAGE_PASSES("Backstage passes to a TAFKAL80ETC concert"),
    SULFURAS ("Sulfuras, Hand of Ragnaros")
    ;

    private final String name;

    ItemType(String name) {
        this.name = name;
    }

    public static ItemType findByValue(String itemName) {
        return Arrays.stream(values())
            .filter(itemType -> itemType.getName().equalsIgnoreCase(itemName))
            .findFirst()
            .orElse(STANDARD);
    }

    public String getName(){
        return this.name;
    }
}
